### PR TITLE
feat: emit Product processor metrics

### PR DESCRIPTION
**Pull request by mediuminvader**
Emit Join metrics on Prometheus endpoint

```

# HELP product_unsatisfied_joins Operations not matching the Join condition
# TYPE product_unsatisfied_joins counter
product_unsatisfied_joins 265

# TYPE product_output_operations counter
product_output_operations 1887841

# TYPE product_input_operations counter
product_input_operations 1888106

# HELP product_right_lookup_size Total number of items in the right lookup table
# TYPE product_right_lookup_size gauge
product_right_lookup_size 265

# HELP product_left_lookup_size Total number of items in the left lookup table
# TYPE product_left_lookup_size gauge
product_left_lookup_size 254

# HELP product_latency Processing latency
# TYPE product_latency summary
product_latency{quantile="0"} 0.0000025
product_latency{quantile="0.5"} 0.000040480100968151095
product_latency{quantile="0.9"} 0.000040480100968151095
product_latency{quantile="0.95"} 0.000040480100968151095
product_latency{quantile="0.99"} 0.000040480100968151095
product_latency{quantile="0.999"} 0.00014963672503375144
product_latency{quantile="1"} 0.028400292
product_latency_sum 17.38265707894686
product_latency_count 1888365

```

### CODE REVIEW

File: dozer-sql/src/pipeline/product/join/processor.rs [hunk 1]
```diff
@@ -16,6 +17,28 @@ pub struct ProductProcessor {
...
+        describe_gauge!(
+            "product.left_lookup_size",
+            "Total number of items in the left lookup table"
+        );
+        describe_gauge!(
+            "product.right_lookup_size",
+            "Total number of items in the right lookup table"
+        );
+        describe_gauge!(
+            "product.unsatisfied_joins",
```
Comment (chubei) [inline dozer-sql/src/pipeline/product/join/processor.rs:29]: Looks like this can only increase? In that case we can use a counter

**Review by chubei — CHANGES_REQUESTED** [review]

File: dozer-sql/src/pipeline/product/join/processor.rs (hunk 1 above)
Comment (mediuminvader) [reply dozer-sql/src/pipeline/product/join/processor.rs:29]: lookup table can decrease in case of TTL eviction

Comment (chubei) [reply dozer-sql/src/pipeline/product/join/processor.rs:29]: I'm talking about `unsatisfied_joins`

**Review by chubei — APPROVED** [review]
Do you think we need labels to distinguish between multiple joins?

**Pull request merged**